### PR TITLE
[Edit Mode] #635 Removed Edit-related concerns from ContextMenuGesture

### PR DESCRIPTION
**Pull request by akhenry**
Addresses #635 and #631

### Summary of Changes
1. Removed edit-specific code from ContextMenuGesture and ActionCapability
2. Created a new EditContextualActionPolicy
3. Used new policy to filter actions attached to scope in ContextMenuController

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

### CODE REVIEW

File: platform/commonUI/general/src/controllers/ContextMenuController.js [hunk 1]
```diff
@@ -36,12 +36,19 @@ define(
...
+
             // Refresh variable "menuActions" in the scope
             function updateActions() {
-                $scope.menuActions = $scope.action ?
-                        $scope.action.getActions({ category: 'contextual' }) :
-                        [];
+                var actions = $scope.action ?
+                    $scope.action.getActions({ category: 'contextual' }) : [];
+
+                $scope.menuActions = actions.filter(applyPolicy);
```
Comment (VWoeltjen) [inline platform/commonUI/general/src/controllers/ContextMenuController.js]: Is this filtering necessary? Policy is enforced by an [`actionService` decorator](https://github.com/nasa/openmct/blob/d6fe543c16ba32c637b2106aaf3da07e0e49929b/platform/policy/src/PolicyActionDecorator.js#L52), so I'd expect these to already be absent.

Comment (akhenry) [inline platform/commonUI/general/src/controllers/ContextMenuController.js]: @VWoeltjen Apparently not :smiley: Didn't even realize there was a decorator being applied here!

**Comment by VWoeltjen** [comment]
I like that this consolidates both issues into a single rule - it's more reflective of the design rationale. 
This addresses #631 well. As for #635, there are still edit-related concerns in [ContextMenuGesture](https://github.com/nasa/openmct/blob/fcc3752ba24e578d8af91f7b40243103e538b5b0/platform/representation/src/gestures/ContextMenuGesture.js#L49-L54) that aren't addressed here.

I'm also leaning toward waiting until next sprint to merge this, since we'll have limited test time and changes are non-user-facing. These changes add maintainability value but I don't think it costs us much to defer merge by a week or so. Thoughts?

**Comment by akhenry** [comment]
@VWoeltjen yeah, there's no rush on this.

**Comment by akhenry** [comment]
> As for #635, there are still edit-related concerns in ContextMenuGesture that aren't addressed here.

Ugh, yep. They're totally unnecessary too, I implemented the fix and forgot to remove the old code. That's what happens when you try to rush something through at night when you're tired. Apologies. I'll push a fix ASAP.

**Comment by akhenry** [comment]
@VWoeltjen Issues resolved, all yours.

### Changes
Reverted `ContextMenuGesture`, `ContextMenuController`, and `ActionCapability` to pre-NEM state.

**Comment by VWoeltjen** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y

Moving to Heinlein, so we can merge then.

**Comment by akhenry** [comment]
After discussion, will defer to Heinlein. Requires some refactoring after changes to context menu handling from #775. Also, it was decided in sprint planning meeting on 20160329 that open in new tab should be available from context menus in edit mode.

**Comment by VWoeltjen** [comment]
Reassigning to resolve merge conflict. If conflict resolution is trivial (e.g. diff is not changed, or is reduced in size due to analogous changes from other issues) go ahead and merge, otherwise reassign to me for re-review

**Comment by akhenry** [comment]
### Changes
1. Resolved merge conflict
2. Added "open in new tab" to allowed actions
3. Fixed failing tests.

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

**Comment by akhenry** [comment]
Addresses #794

**Comment by VWoeltjen** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y

**Pull request merged**